src/sanitize/sanitizeNumbers.ts

```diff
@@ -146,6 +146,15 @@ export function sanitizeNumbers(data: unknown): AnyJson {
 		return mapTypeSanitizeKeyValue(data);
 	}
 
+	/**
+	 * When the given data has a superclass of `BN`, we need to check the 
+	 * constructor to see if the data is a subclass of `BN` since `instanceof` 
+	 * will not check the complete prototype chain.
+	 */
+	if (typeof data === 'object' && data && data.constructor.name === 'BN') {
```

Comment:
Instead of comparing string name you could just do `data.constructor === BN`. I'm still not sure if this is the correct check here necessarily, but if it works in practice it works.

Reply:
Weird, but oh well.

Reply:
For documentation sake: Fixed via setting a resolution on `bn.js`